daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/unparsers/UState.scala

```diff
@@ -343,14 +342,6 @@ abstract class UState(
 
   def documentElement: DIDocument
 
-  def newVariableInstance(vrd: VariableRuntimeData): Unit = {
-    variableMap.newVariableInstance(vrd, this)
-  }
-
-  def removeVariableInstance(vrd: VariableRuntimeData): Unit = {
-    variableMap.removeVariableInstance(vrd)
-  }
```

Comment:
I was expecting to a see a change in this file when we clone the ustate for suspensions. I think based on a discussion on the mailing list, the NVIUnparser should always create a newVariableInstance and then create a suspension if a defautlValue is needed to be set (since that could suspend).

Then if anything else suspends (like reading this NVI default value), then when clongin the ustate for suspension, we need to create a shallow copy of the top of the variable map. This way the suspensions created while this NVI was in scope have pointer to that instance. But because it's a shallow copy if one of those suspensions mutates the state (i.e. marking the variable as read) other suspension will see this new state.